command/transform/transformation.go

```diff
@@ -56,7 +58,7 @@ func (t *Transformation) Run(parser *dockerfile.Parser) error {
 	}
 
 	file.From.Emit(t.Output)
-	return t.write(parser, &file, "")
+	return t.write(parser, &file, "", map[string]bool{file.From.Image: true})
```

Comment:
I think we should introduce a separate type for the list of provided images within the `provided` package. Will make it more clear what is happening here, and allow to hide some of the specific details of the handling in the `write()` method below.